WordPress/gutenberg
# Add non-listbox functionality back to `CircularOptionPicker`

## What?
This PR puts back the ability to use `CircularOptionPicker` as buttons, which was removed in #52255, while maintaining the new (now default) `listbox` behaviour. (Closes #54239.)

## Why?
In #35292, `CircularOptionPicker` was refactored to present better semantics and keyboard navigation for picking options. However, as per #54239, it was discovered that `CircularOptionPicker` is not just used as an "option picker", but also as an access method for editing options. As such, we need to put the original functionality back in place, while maintaining the new functionality, so that the control can be used in either way.

## How?
A new `asButtons` flag has been added to `CircularOptionPicker`, allowing component consumers to decide how it should be rendered, depending on context. This is passed on by `ColorPalette`, `DuotonePicker` and `GradientPicker`, the higher-order components that expose it.

## Testing Instructions
Superficially, nothing should change, and `CircularOptionPicker` should continue to behave as before. Clicking each option should fire an `onClick` event, and in Storybook, the selected option should show as checked. Toggling `asButtons` shouldn't impact this behaviour.

### Testing Instructions for Keyboard
With a keyboard, the default behaviour should also continue as before. The `CircularOptionPicker` control should still present as a single tab stop, and arrow keys should navigate between the different options. Semantically, it should still present as a `listbox` with multiple `option`s, marked `aria-selected` as appropriate.

Enabling `asButtons` should change this, so that each option becomes its own tab stop. The `listbox` semantics should be removed, and each option should be a `button`, marked `aria-pressed` where needed.
